components/ws-manager-bridge/src/cluster-service-server.ts

```diff
@@ -155,10 +156,19 @@ export class ClusterService implements IClusterServiceServer {
                                 ),
                             );
                         } else {
-                            resolve();
+                            resolve(resp);
                         }
                     });
                 });
+                newCluster.preferredWorkspaceClass = clusterDesc.getPreferredWorkspaceClass();
+                newCluster.availableWorkspaceClasses = clusterDesc.getWorkspaceClassesList().map((c) => {
+                    return <WorkspaceClass>{
+                        creditsPerMinute: c.getCreditsPerMinute(),
+                        description: c.getDescription(),
+                        displayName: c.getDisplayName(),
+                        id: c.getId(),
+                    };
+                });
```

Comment:
I believe we could remove the duplicate logic here and rely on the update logic in bridge.start if it were executed eagerly.

Reply:
This is the connection check to verify that we can actually connect to the workspace cluster. We don't want to put a "broken connection" on the record.

Before this PR this was a "GetWorkspace" call for the same reason.

I'll add a comment.

Reply:
Yes, I understand. I'm just concerned about the duplicate update logic as it might drift apart and generally complicates things if there's not a single place responsible for this.